calculator.py

```diff
@@ -0,0 +1,157 @@
+# REQUIREMENTS
+# -> to be used in the command line
+# -> operations: add, subtract, divide, multiply
+
+# ACTION PLAN
+# [x] decide on the input criteria
+# [x] draft intro text
+# [x] gather imput funtion(s)
+# [x] eveluating / converting input
+#       [x] choose which operation
+#       [x] extract the numbers
+# [x] making calculations
+#       [] write calculation function(s)
+#       [] call the right funtion
+# [x] displaying result
+# [x] create separate funtions for operations -> not sure if that makes sense, the code is not simpler and the advantage for testing might be minor
+# [x] add testing 
+# [x] finalise intro text, remove test statements from intro
+# [x] make the calculator run without closing until a command for closing the programm is given
+# [x] add more error and input handling: letters, other input mistakes 
+# [x] handle negative numbers
+# [] check if possible to remove letters to fix the input (after or before eval)
+# [] check possibility to rewrite simpler wiht compile /eval funtions only
+
+# Input criteria
+# -> one line / one input
+# -> start with 2 numbers 
+# -> 1 operation
+
+import time
+
+def instruction():
+    welcome_message = "Welcome to the calculator. \n----------\nPossible operations: \naddition \t-> + \nsubtraction \t-> - \nmultiplication \t-> * \ndivision \t-> / \n----------\nQUIT: q\n----------"
+    print(welcome_message)
+    return
+
+def convert(user_input):
+    invalid_number_warning = "Please enter numbers."
+    operator_warning = "You did not input a valid operator."
+    possible_operations = ["+", "*", "/", "-"]
+
+    x = 0 
+    y = 0
+    requested_operation = ""
+
+    for el in possible_operations: 
+        operator_index = user_input.find(el)
+
+        if operator_index == -1: 
+            continue
+
+        elif operator_index > 0: 
+            requested_operation = el
+
+            try: 
+                x = eval(user_input[0:operator_index])
+                y = eval(user_input[operator_index+1:])
```

Comment:
I don't think you should be using `eval`. It has much more effect than type conversion. For example try this test case:
```
def test_extended_eval(self):
        self.assertEqual(convert("len('hello world')+sum(range(10))"), (11, 45, "+"))
```

Notice it passes instead of returning an error.

What you want to use here is something like `int(user_input[0:operator_index])` or `float(user_input[0:operator_index])`